crates/swc_plugin_comments/src/plugin_comments_proxy.rs

```diff
@@ -81,54 +111,135 @@ impl Comments for PluginCommentsProxy {
             .expect("Should able to deserialize CommentsVecPtr")
         };
 
-        // Using CommentsVecPtr's value, reconstruct actual Vec<Comments>
+        // Using CommentsVecPtr's value, reconstruct actual Comment, or Vec<Comments>
         Some(unsafe {
             Serialized::deserialize_from_ptr(comments_vec_ptr.0 as _, comments_vec_ptr.1)
                 .expect("Returned comments should be serializable")
         })
     }
+}
+
+#[cfg(feature = "plugin-mode")]
+#[cfg_attr(not(target_arch = "wasm32"), allow(unused))]
+impl Comments for PluginCommentsProxy {
+    fn add_leading(&self, pos: BytePos, cmt: Comment) {
+        self.allocate_comments_buffer_to_host(&cmt);
+        #[cfg(target_arch = "wasm32")]
+        unsafe {
+            __add_leading_comment_proxy(pos.0);
```

Comment:
How does this work? We don't pass the pointer of the host memory we allocated above?

Reply:
we do, https://github.com/swc-project/swc/pull/4229/files#diff-5d0fdaffb501760b81888c7b39468d14561cb4f509dd056f2f68ebdfaf3ead93R45

right above does the job. this is same way to set tranaform result, using hostenv's attached buffer guest copies value into host.

Reply:
Yeah, but I mean I'm not sure how `__add_leading_comment_proxy` works without `&pointer_to_comment_in_host_memory`

Reply:
Ah, I found `mutable_comment_buffer`